### PR TITLE
Support single line comments of the form `// comment`

### PR DESCRIPTION
**Pull request by IwanKaramazow**
Implements parsing & printing of single line comments.

Example:
```reason
// single line comment
let x = 1;
```

The main logic is taken from https://github.com/facebook/reason/pull/1800.
Extra test cases and support for whitespace interleaving have been implemented.

### CODE REVIEW

File: formatTest/unit_tests/expected_output/basicStructures.re [hunk 1]
```diff
@@ -146,7 +146,7 @@ let x = arr^[0] = 1;
  *============================================================================
  */;
 
-let (/\/) = (+); /* // is not a comment */
+let (/++) = (+); /* // is not a comment */
```
Comment (kpsuperplane) [inline formatTest/unit_tests/expected_output/basicStructures.re]: Forgive me I'm pretty new to Reason but, I think the comment about `// is not a comment` should be updated?

File: formatTest/unit_tests/input/basicStructures.re [hunk 2]
```diff
@@ -134,7 +134,7 @@ let x = arr^[0] = 1;
  *============================================================================
  */;
 
-let (//) = (+); /* // is not a comment */
+let (/++) = (+); /* // is not a comment */
```
Comment (kpsuperplane) [inline formatTest/unit_tests/input/basicStructures.re]: Ditto on comment about not a comment

File: formatTest/unit_tests/expected_output/basicStructures.re (hunk 1 above)
Comment (IwanKaramazow) [reply formatTest/unit_tests/expected_output/basicStructures.re]: Thanks! Updated.

File: formatTest/unit_tests/input/basicStructures.re (hunk 2 above)
Comment (IwanKaramazow) [reply formatTest/unit_tests/input/basicStructures.re]: Also updated.
